lib/generator/code_generator.dart

```diff
@@ -10,83 +10,109 @@ import 'package:graphql_generator/generator/objects.dart';
 import 'package:graphql_generator/generator/union.dart';
 
 class GraphQLCodeGenerators {
-  static final GraphQLCodeGenerators _singleton = new GraphQLCodeGenerators
-      ._internal();
-
-  factory GraphQLCodeGenerators() {
-    return _singleton;
-  }
-
-  GraphQLCodeGenerators._internal();
-
-  String namespace;
-  Map<String, DartType> types = {};
-  Map<String, String> fragments = {};
-
-  List<ObjectType> responseTypes;
-  List<ObjectType> enumTypes = [];
-  List<ObjectType> interfaceTypes = [];
-  List<ObjectType> unionTypes = [];
-  List<ObjectType> objectTypes = [];
-  List<ObjectType> inputObjectTypes = [];
-
-  ObjectType mutation;
-  String mutationClassName;
-
-  Map<String, Class> classes = {};
-
-
-  graphQLGenerate(List<ObjectType> responseTypes,
+  generate(
+      List<ObjectType> responseTypes,
       String namespace,
       Map<String, DartType> types,
       Map<String, String> fragments,
       String mutationClassName) {
-    this.namespace = namespace;
-    this.responseTypes = responseTypes;
-    this.types = types;
-    this.fragments = fragments;
-    this.mutationClassName = mutationClassName;
-    splitTypes();
-    return generateTypes();
+    List<ObjectType> enumObjectTypes =
+        getTypesFromList(responseTypes, Kind.ENUM);
+    List<ObjectType> interfaceObjectTypes =
+        getTypesFromList(responseTypes, Kind.INTERFACE);
+    List<ObjectType> unionObjectTypes =
+        getTypesFromList(responseTypes, Kind.UNION);
+    List<ObjectType> objectObjectTypes =
+        getTypesFromList(responseTypes, Kind.OBJECT);
+    List<ObjectType> inputObjectTypes =
+        getTypesFromList(responseTypes, Kind.INPUT_OBJECT);
+    ObjectType mutationObjectType =
+        getMutationObjectType(responseTypes, mutationClassName);
+
+    return generateTypes(
+        namespace,
```

Comment:
@justinjoyn I have an idea here. What if you didn't do any filtering here and instead passed all object types to all generators? The generators would be responsible for filtering the type (like `responseTypes.where((type) => type.kind == typeKind)`. This way you don't have to pass so many arguments around.